File: lib_network_retrofit/src/main/java/com/network/retrofit/api/NetworkServiceManager.java
package com.network.retrofit.api;

import com.network.retrofit.OnRequestListener;
import com.network.retrofit.RetrofitClient;
import com.network.retrofit.ServiceHelper;
import com.network.retrofit.api.data.MovieBean;
import com.network.retrofit.response.NetworkBaseResponse;


import java.util.List;

import retrofit2.Call;

public class NetworkServiceManager {

    private static INetworkService networkService = RetrofitClient.getInstance().create(INetworkService.class);

    /**
     * 影片列表
     * @param param
     * @param listener
     * @return
     */
    public static Call movieList(String param, OnRequestListener<List<MovieBean>> listener) {
        Call<NetworkBaseResponse> call = networkService.movieList(param);
        ServiceHelper.callEntities(call, MovieBean.class, listener);
        return call;

    }

    /**
     * 影片详情
     * @param param
     * @param listener
     * @return
     */
    public static Call movieDetail(String param, OnRequestListener<MovieBean> listener) {
        Call<NetworkBaseResponse> call = networkService.mediaDetail(param);
        ServiceHelper.callEntity(call, MovieBean.class, listener);
        return call;

    }


}

File: lib_network_retrofit_coroutine/src/main/java/com/network/retrofit_coroutine/api/NetworkServiceManager.kt
package com.network.retrofit_coroutine.api

import com.network.retrofit_coroutine.RetrofitClient
import com.network.retrofit_coroutine.api.data.VersionMsgBean
import com.network.retrofit_coroutine.response.NetworkBaseResponse
import kotlinx.coroutines.Deferred

object NetworkServiceManager {

    private val serviceApi: NetworkServiceApi =
        RetrofitClient.getInstance().create(NetworkServiceApi::class.java)

    fun checkVersionDeferred(): Deferred<NetworkBaseResponse<VersionMsgBean?>> {
        return serviceApi.checkVersion_deferred()
    }

    suspend fun checkVersionSuspend(): VersionMsgBean {
        val response = serviceApi.checkVersion_suspend();
        if (response.code == 0) {
            throw Exception("code is ${response.code}")
        }
        return response.data ?: throw Exception("data is null")
    }
}